components/director/internal/domain/formationassignment/service.go

```diff
@@ -0,0 +1,170 @@
+package formationassignment
+
+import (
+	"context"
+
+	"github.com/kyma-incubator/compass/components/director/internal/domain/tenant"
+	"github.com/kyma-incubator/compass/components/director/internal/model"
+	"github.com/kyma-incubator/compass/components/director/pkg/apperrors"
+	"github.com/kyma-incubator/compass/components/director/pkg/log"
+	"github.com/kyma-incubator/compass/components/director/pkg/resource"
+	"github.com/pkg/errors"
+)
+
+// FormationAssignmentRepository represents the formation assignment repository layer
+//go:generate mockery --name=FormationAssignmentRepository --output=automock --outpkg=automock --case=underscore --disable-version-string
+type FormationAssignmentRepository interface {
+	Create(ctx context.Context, item *model.FormationAssignment) error
+	Get(ctx context.Context, id, tenantID string) (*model.FormationAssignment, error)
+	GetForFormation(ctx context.Context, tenantID, id, formationID string) (*model.FormationAssignment, error)
+	List(ctx context.Context, pageSize int, cursor, tenantID string) (*model.FormationAssignmentPage, error)
+	ListByFormationIDs(ctx context.Context, tenantID string, formationIDs []string, pageSize int, cursor string) ([]*model.FormationAssignmentPage, error)
+	Update(ctx context.Context, model *model.FormationAssignment) error
+	Delete(ctx context.Context, id, tenantID string) error
+	Exists(ctx context.Context, id, tenantID string) (bool, error)
+}
+
+// UIDService generates UUIDs for new entities
+//go:generate mockery --name=UIDService --output=automock --outpkg=automock --case=underscore --disable-version-string
+type UIDService interface {
+	Generate() string
+}
+
+type service struct {
+	repo   FormationAssignmentRepository
+	uidSvc UIDService
+}
+
+// NewService creates a FormationTemplate service
+func NewService(repo FormationAssignmentRepository, uidSvc UIDService) *service {
+	return &service{
+		repo:   repo,
+		uidSvc: uidSvc,
+	}
+}
+
+// Create creates a formation assignment using `in`
+func (s *service) Create(ctx context.Context, in *model.FormationAssignmentInput) (string, error) {
+	formationAssignmentID := s.uidSvc.Generate()
+	tenantID, err := tenant.LoadFromContext(ctx)
+	if err != nil {
+		return "", errors.Wrapf(err, "while loading tenant from context")
+	}
+	log.C(ctx).Debugf("ID: %q generated for formation assignment for tenant with ID: %q", formationAssignmentID, tenantID)
+
+	log.C(ctx).Infof("Creating formation assignment with source: %q and source type: %q, and target: %q with target type: %q", in.Source, in.SourceType, in.Target, in.TargetType)
```

Comment:
Maybe these types of logs will be helpful in the other functions too(Delete, Update, etc.).